src/stan/model/indexing/lvalue.hpp

```diff
@@ -356,43 +718,196 @@ inline std::enable_if_t<!std::is_same<I, index_uni>::value> assign(
+ * @param[in] y Matrix variable to assign from.
+ * @param[in] name Name of variable (default "ANON").
+ * @param[in] depth Indexing depth (default 0).
+ * @throw std::out_of_range If any of the indices are out of bounds.
+ * @throw std::invalid_argument If the dimensions of the indexed
+ * matrix and right-hand side matrix do not match.
+ */
+template <typename Mat1, typename Mat2, typename Idx,
+          stan::internal::require_eigen_dense_dynamic_t<Mat1>* = nullptr>
+inline void assign(
+    Mat1&& x,
+    const cons_index_list<Idx, cons_index_list<index_min_max, nil_index_list>>&
+        idxs,
+    const Mat2& y, const char* name = "ANON", int depth = 0) {
+  if (idxs.tail_.head_.is_ascending()) {
+    const auto col_start = idxs.tail_.head_.min_ - 1;
+    const auto col_size = idxs.tail_.head_.max_ - col_start;
+    stan::math::check_range("matrix[..., min_max] assign range", name, x.cols(),
+                            idxs.tail_.head_.min_);
+    stan::math::check_range("matrix[..., min_max] assign range", name,
+                            idxs.tail_.head_.max_, x.cols());
+    stan::math::check_size_match("matrix[..., min_max] assign col size", "lhs",
+                                 idxs.tail_.head_.max_, name, x.cols());
+    assign(x.middleCols(col_start, col_size), index_list(idxs.head_), y, name,
+           depth + 1);
+    return;
+  } else {
+    const auto col_start = idxs.tail_.head_.max_ - 1;
+    const auto col_size = idxs.tail_.head_.min_ - col_start;
+    stan::math::check_range("matrix[..., reverse_min_max] assign range", name,
+                            x.cols(), idxs.tail_.head_.max_);
+    stan::math::check_range("matrix[..., reverse_min_max] assign range", name,
+                            idxs.tail_.head_.min_, x.cols());
+    stan::math::check_size_match("matrix[..., min_max] assign col size", "lhs",
+                                 idxs.tail_.head_.min_, name, x.cols());
+    assign(x.middleCols(col_start, col_size), index_list(idxs.head_),
+           y.rowwise().reverse(), name, depth + 1);
+    return;
   }
 }
 
 /**
- * Assign the specified array (standard vector) at the specified
- * index list beginning with a single index to the specified value.
+ * Assign the elements of one standard vector to another.
  *
- * This function operates recursively to carry out the tail
- * indexing.
+ *  std_vector = std_vector
  *
- * Types:  x[uni | L] = y
+ * @tparam T Type of Std vector to be assigned to.
+ * @tparam U Type of Std vector to be assigned from.
+ * @param[in] x lvalue variable
+ * @param[in] y rvalue variable
+ * @param[in] name name of lvalue variable (default "ANON").
+ * @param[in] depth indexing depth (default 0).
+ */
+template <typename T, typename U, require_all_std_vector_t<T, U>* = nullptr,
+          require_not_t<
+              std::is_assignable<std::decay_t<T>&, std::decay_t<U>>>* = nullptr>
+inline void assign(T&& x, const nil_index_list& /* idxs */, U&& y,
```

Comment:
Actually why don't we forward these along? Since one (`x[uni] = y`) is just doing assignment and depending on the other (`std_vector<T>[uni | Idx] = T`) has the potential to use a move depending on the tail index (i.e. like the right index being an omni `A[1, :] = B`)

Reply:
nvm I just saw your other comment!